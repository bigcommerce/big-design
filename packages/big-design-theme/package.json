{
  "name": "@bigcommerce/big-design-theme",
  "version": "1.1.0",
  "sideEffects": false,
  "main": "src/index.ts",
  "license": "(MIT AND CC-BY-3.0)",
  "author": "BigCommerce Inc",
  "repository": {
    "type": "git",
    "url": "https://github.com/bigcommerce/big-design.git",
    "directory": "packages/big-design-theme"
  },
  "scripts": {
    "build": "pnpm run build:cjs && pnpm run build:es && pnpm run build:dt",
    "build:cjs": "NODE_ENV=production BABEL_ENV=cjs babel --extensions \".ts,.tsx\" ./src --out-dir ./dist/cjs",
    "build:es": "NODE_ENV=production BABEL_ENV=es babel --extensions \".ts,.tsx\" ./src --out-dir ./dist/es",
    "build:dt": "tsc -p tsconfig.declarations.json --emitDeclarationOnly",
    "lint": "eslint . --max-warnings 0",
    "prepack": "big-design-prepack",
    "postpack": "big-design-postpack",
    "test": "jest",
    "test:watch": "pnpm run test --watch",
    "typecheck": "tsc --noEmit"
  },
  "files": [
    "dist"
  ],
  "publishConfig": {
    "access": "public",
    "main": "dist/cjs/index.js",
    "module": "dist/es/index.js",
    "typings": "dist/index.d.ts"
  },
  "prettier": "@bigcommerce/eslint-config/prettier",
  "dependencies": {
    "@babel/runtime": "^7.26.0",
    "polished": "^4.3.1"
  },
  "peerDependencies": {
    "react": "^18.0.0",
    "react-dom": "^18.0.0",
    "styled-components": "^5.3.5"
  },
  "devDependencies": {
    "@babel/cli": "^7.25.7",
    "@babel/core": "^7.26.0",
    "@babel/plugin-transform-runtime": "^7.25.9",
    "@babel/preset-env": "^7.26.0",
    "@babel/preset-react": "^7.26.3",
    "@babel/preset-typescript": "^7.26.0",
    "@bigcommerce/configs": "workspace:^",
    "@bigcommerce/pack": "workspace:^",
    "@styled/typescript-styled-plugin": "^1.0.1",
    "@swc/core": "^1.10.4",
    "@swc/jest": "^0.2.37",
    "@swc/plugin-styled-components": "^6.0.2",
    "@types/jest": "^29.5.14",
    "@types/styled-components": "^5.1.34",
    "babel-jest": "^29.0.2",
    "babel-plugin-styled-components": "^2.0.7",
    "jest": "^29.7.0",
    "jest-environment-jsdom": "^29.4.1",
    "jest-fail-on-console": "^3.3.1",
    "jest-styled-components": "^7.1.1",
    "react": "^18.2.0",
    "react-dom": "^18.2.0",
    "styled-components": "^5.3.11",
    "typescript": "^5.6.3"
  }
}
