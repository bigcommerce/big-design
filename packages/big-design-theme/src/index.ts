import { createHelpers, Helpers } from './helpers';
import { ThemeOptions, themeOptions } from './options';
import { Border, BorderRadius, createBorder, createBorderRadius } from './system/border';
import { Breakpoints, breakpoints, BreakpointValues, breakpointValues } from './system/breakpoints';
import { Colors, colors } from './system/colors';
import * as keyframes from './system/keyframes';
import { createLineHeight, LineHeight } from './system/line-height';
import { Shadow, shadow } from './system/shadow';
import { createSpacing, Spacing } from './system/spacing';
import { createTypography, Typography } from './system/typography';
import { ZIndex, zIndex } from './system/z-index';

export * from './helpers';
export * from './system';

export interface ThemeInterface {
  border: Border;
  borderRadius: BorderRadius;
  breakpointValues: BreakpointValues;
  breakpoints: Breakpoints;
  colors: Colors;
  helpers: Helpers;
  keyframes: typeof keyframes;
  lineHeight: LineHeight;
  shadow: Shadow;
  spacing: Spacing;
  typography: Typography;
  zIndex: ZIndex;
}

export const createTheme = (customOptions: Partial<ThemeOptions> = {}): ThemeInterface => {
  themeOptions.setOptions(customOptions);

  return {
    border: createBorder(),
    borderRadius: createBorderRadius(),
    breakpointValues,
    breakpoints,
    colors,
    helpers: createHelpers(),
    keyframes,
    lineHeight: createLineHeight(),
    shadow,
    spacing: createSpacing(),
    typography: createTypography(),
    zIndex,
  };
};

export const theme: ThemeInterface = createTheme();
