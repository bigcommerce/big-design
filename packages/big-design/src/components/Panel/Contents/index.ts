export { PanelContents, type PanelContentProps } from './Contents';
