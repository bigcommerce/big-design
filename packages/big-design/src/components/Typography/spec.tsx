import { theme } from '@bigcommerce/big-design-theme';
import { render } from '@testing-library/react';
import React from 'react';

import 'jest-styled-components';

import { H0, H1, H2, H3, H4, HR, Small, Text } from './Typography';

test('render H0', () => {
  const { container } = render(<H0>Test</H0>);

  expect(container.firstChild).toMatchSnapshot();
});

test('render H1', () => {
  const { container } = render(<H1>Test</H1>);

  expect(container.firstChild).toMatchSnapshot();
});

test('render H2', () => {
  const { container } = render(<H2>Test</H2>);

  expect(container.firstChild).toMatchSnapshot();
});

test('render H3', () => {
  const { container } = render(<H3>Test</H3>);

  expect(container.firstChild).toMatchSnapshot();
});

test('render H4', () => {
  const { container } = render(<H4>Test</H4>);

  expect(container.firstChild).toMatchSnapshot();
});

test('render Small', () => {
  const { container } = render(<Small>Test</Small>);

  expect(container.firstChild).toMatchSnapshot();
});

test('render Text', () => {
  const { container } = render(<Text>Test</Text>);

  expect(container.firstChild).toMatchSnapshot();
});

test('render HR', () => {
  const { container } = render(<HR />);

  expect(container.firstChild).toMatchSnapshot();
});

test('render Text with ellipsis', () => {
  const { container } = render(<Text ellipsis={true}>Test with ellipsis</Text>);

  expect(container.firstChild).toMatchSnapshot();
});

test('H0 - does not forward styles', () => {
  const { container } = render(
    <H0 className="test" style={{ background: 'red' }}>
      Test
    </H0>,
  );

  expect(container.getElementsByClassName('test')).toHaveLength(0);
  expect(container.firstChild).not.toHaveStyle('background: red');
});

test('H1 - does not forward styles', () => {
  const { container } = render(
    <H1 className="test" style={{ background: 'red' }}>
      Test
    </H1>,
  );

  expect(container.getElementsByClassName('test')).toHaveLength(0);
  expect(container.firstChild).not.toHaveStyle('background: red');
});

test('H2 - does not forward styles', () => {
  const { container } = render(
    <H2 className="test" style={{ background: 'red' }}>
      Test
    </H2>,
  );

  expect(container.getElementsByClassName('test')).toHaveLength(0);
  expect(container.firstChild).not.toHaveStyle('background: red');
});

test('H3 - does not forward styles', () => {
  const { container } = render(
    <H3 className="test" style={{ background: 'red' }}>
      Test
    </H3>,
  );

  expect(container.getElementsByClassName('test')).toHaveLength(0);
  expect(container.firstChild).not.toHaveStyle('background: red');
});

test('H4 - does not forward styles', () => {
  const { container } = render(
    <H4 className="test" style={{ background: 'red' }}>
      Test
    </H4>,
  );

  expect(container.getElementsByClassName('test')).toHaveLength(0);
  expect(container.firstChild).not.toHaveStyle('background: red');
});

test('Small - does not forward styles', () => {
  const { container } = render(
    <Small className="test" style={{ background: 'red' }}>
      Test
    </Small>,
  );

  expect(container.getElementsByClassName('test')).toHaveLength(0);
  expect(container.firstChild).not.toHaveStyle('background: red');
});

test('Text - does not forward styles', () => {
  const { container } = render(
    <Text className="test" style={{ background: 'red' }}>
      Test
    </Text>,
  );

  expect(container.getElementsByClassName('test')).toHaveLength(0);
  expect(container.firstChild).not.toHaveStyle('background: red');
});

test('HR - does not forward styles', () => {
  const { container } = render(<HR className="test" style={{ background: 'red' }} />);

  expect(container.getElementsByClassName('test')).toHaveLength(0);
  expect(container.firstChild).not.toHaveStyle('background: red');
});

test('All typography components allow changing their color given a color prop', () => {
  const { container } = render(
    <>
      <H1 color="primary">Test</H1>
      <H2 color="primary">Test</H2>
      <H3 color="primary">Test</H3>
      <H4 color="primary">Test</H4>
      <Small color="primary">Test</Small>
      <Text color="primary">Test</Text>
    </>,
  );

  Array.from(container.children).forEach((child) => {
    expect(child).toHaveStyle(`color: ${theme.colors.primary}`);
  });
});

test('HR allows changing its color given a color prop', () => {
  const { container } = render(<HR color="primary" />);

  expect(container.firstChild).toHaveStyle(`border-bottom: 1px solid ${theme.colors.primary}`);
});

test('HR can change its margins given a margin prop', () => {
  const { container } = render(<HR marginTop="medium" />);

  expect(container.firstChild).toHaveStyle(`margin-top: ${theme.spacing.medium}`);
});

test('Headings can change their tag', () => {
  const { container } = render(<H2 as="h1">Test</H2>);

  expect(container.querySelector('h1')).toBeInTheDocument();
});

test('Headings can not change their tag to non-heading tags', () => {
  // @ts-expect-error ignoring since as="p" is not a valid prop
  const { container } = render(<H2 as="p">Test</H2>);

  expect(container.querySelector('p')).not.toBeInTheDocument();
  expect(container.querySelector('h2')).toBeInTheDocument();
});

test('Text and Small can change their tag', () => {
  const { container } = render(
    <>
      <Text as="span">Some Text</Text>
      <Small as="span">Some Text</Small>
    </>,
  );

  expect(container.querySelectorAll('span')).toHaveLength(2);
});

test('Text and Small accept text modifiers', () => {
  const { getByTestId } = render(
    <>
      <Text bold data-testid="text" italic underline uppercase>
        Some Text
      </Text>
      <Small data-testid="small" lowercase strikethrough>
        Some Text
      </Small>
    </>,
  );

  const text = getByTestId('text');
  const small = getByTestId('small');

  expect(text).toHaveStyle(`
    font-weight: 600;
    font-style: italic;
    text-decoration: underline;
    text-transform: uppercase;
  `);

  expect(small).toHaveStyle(`
    text-decoration: line-through;
    text-transform: lowercase;
  `);
});
