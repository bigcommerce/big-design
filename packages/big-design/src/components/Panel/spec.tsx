import { theme } from '@bigcommerce/big-design-theme';
import { fireEvent, render, screen } from '@testing-library/react';
import React, { createRef } from 'react';
import 'jest-styled-components';

import { Text } from '../Typography';

import { Panel } from './Panel';

test('render panel', () => {
  render(
    <Panel data-testid="panel">
      Dolore proident eiusmod sint est enim laboris anim minim quis ut adipisicing consectetur
      officia ex. Ipsum eiusmod fugiat amet pariatur culpa tempor aliquip tempor nisi. Irure esse
      deserunt nostrud ipsum id adipisicing enim velit labore. Nulla exercitation laborum laboris
      Lorem irure sit esse nulla mollit aliquip consectetur velit
    </Panel>,
  );

  const panel = screen.getByTestId('panel');

  expect(panel).toMatchSnapshot();
});

test('render panel with only a heading and no content', () => {
  render(<Panel data-testid="panel" header="Test Header" />);

  const panel = screen.getByTestId('panel');

  expect(panel).toMatchSnapshot();
});

test('does not forward styles', () => {
  render(
    <Panel className="test" data-testid="panel" style={{ background: 'red' }}>
      Dolore proident eiusmod sint est enim laboris anim minim quis ut adipisicing consectetur
      officia ex. Ipsum eiusmod fugiat amet pariatur culpa tempor aliquip tempor nisi. Irure esse
      deserunt nostrud ipsum id adipisicing enim velit labore. Nulla exercitation laborum laboris
      Lorem irure sit esse nulla mollit aliquip consectetur velit
    </Panel>,
  );

  const panel = screen.getByTestId('panel');

  expect(screen.queryByText('test')).not.toBeInTheDocument();
  expect(panel).not.toHaveStyle('background: red');
});

test('renders a header and action', () => {
  render(
    <Panel action={{ text: 'Test Action' }} header="Test Header">
      Dolore proident eiusmod sint est enim laboris anim minim quis ut adipisicing consectetur
      officia ex. Ipsum eiusmod fugiat amet pariatur culpa tempor aliquip tempor nisi. Irure esse
      deserunt nostrud ipsum id adipisicing enim velit labore. Nulla exercitation laborum laboris
      Lorem irure sit esse nulla mollit aliquip consectetur velit
    </Panel>,
  );

  const header = screen.getByRole('heading');

  expect(header).toBeInTheDocument();
  expect(header).toHaveTextContent('Test Header');

  const actionButton = screen.getByRole('button');

  expect(actionButton).toBeInTheDocument();
  expect(actionButton).toHaveTextContent('Test Action');
});

test('renders a badge and header', () => {
  render(
    <Panel badge={{ label: 'danger', variant: 'danger' }} header="Test Header">
      Dolore proident eiusmod sint est enim laboris anim minim quis ut adipisicing consectetur
      officia ex. Ipsum eiusmod fugiat amet pariatur culpa tempor aliquip tempor nisi. Irure esse
      deserunt nostrud ipsum id adipisicing enim velit labore. Nulla exercitation laborum laboris
      Lorem irure sit esse nulla mollit aliquip consectetur velit
    </Panel>,
  );

  const header = screen.getByRole('heading');

  expect(header).toBeInTheDocument();

  const badge = screen.getByText('danger');

  expect(badge).toBeInTheDocument();
});

describe('description', () => {
  test('renders string description', async () => {
    render(
      <Panel action={{ text: 'Test Action' }} description="Test Description" header="Test Header">
        Dolore proident eiusmod sint est enim laboris anim minim quis ut adipisicing consectetur
        officia ex. Ipsum eiusmod fugiat amet pariatur culpa tempor aliquip tempor nisi. Irure esse
        deserunt nostrud ipsum id adipisicing enim velit labore. Nulla exercitation laborum laboris
        Lorem irure sit esse nulla mollit aliquip consectetur velit
      </Panel>,
    );

    const description = await screen.findByText('Test Description');

    expect(description).toBeInTheDocument();
  });

  test('renders ReactNode description', async () => {
    render(
      <Panel
        action={{ text: 'Test Action' }}
        description={<Text>Test Description</Text>}
        header="Test Header"
      >
        Dolore proident eiusmod sint est enim laboris anim minim quis ut adipisicing consectetur
        officia ex. Ipsum eiusmod fugiat amet pariatur culpa tempor aliquip tempor nisi. Irure esse
        deserunt nostrud ipsum id adipisicing enim velit labore. Nulla exercitation laborum laboris
        Lorem irure sit esse nulla mollit aliquip consectetur velit
      </Panel>,
    );

    const description = await screen.findByText('Test Description');

    expect(description).toBeInTheDocument();
  });
});

test('action options get forwarded to button', () => {
  const onClick = jest.fn();

  render(
    <Panel action={{ text: 'Test Action', onClick }} header="Test Header">
      Dolore proident eiusmod sint est enim laboris anim minim quis ut adipisicing consectetur
      officia ex. Ipsum eiusmod
    </Panel>,
  );

  fireEvent.click(screen.getByRole('button'));

  expect(onClick).toHaveBeenCalled();
});

test('forwards data attributes', () => {
  render(
    <Panel data-testid="panel" header="Test Header">
      Dolore proident eiusmod sint est enim laboris
    </Panel>,
  );

  const panel = screen.getByTestId('panel');

  expect(panel).toBeInTheDocument();
});

test('forwards ref', () => {
  const ref = createRef<HTMLDivElement>();

  render(<Panel data-testid="panel" ref={ref} />);

  const panel = screen.getByTestId('panel');

  expect(panel).toBe(ref.current);
});

test('ignores padding props', () => {
  render(
    // @ts-expect-error - ignoring since paddingRight is not a valid prop
    <Panel data-testid="panel" header="Test Header" paddingRight="xxxLarge">
      Dolore proident eiusmod sint est enim laboris
    </Panel>,
  );

  const panel = screen.getByTestId('panel');

  expect(panel).not.toHaveStyle(`padding-right: ${theme.spacing.xxxLarge}`);
});

test("panel action doesn't go to full width", () => {
  render(
    <Panel action={{ text: 'Test Action' }} header="Test Header">
      Test
    </Panel>,
  );

  const button = screen.getByRole('button');

  expect(button).toHaveStyle('width: auto');
});

test('forwards headerId to heading', () => {
  render(
    <Panel header="Test Header" headerId="test-header">
      Test
    </Panel>,
  );

  expect(screen.getByText('Test Header')).toHaveAttribute('id', 'test-header');
});

test('applies the right border radius values', async () => {
  render(
    <Panel role="region">
      Dolore proident eiusmod sint est enim laboris anim minim quis ut adipisicing consectetur
      officia ex. Ipsum eiusmod fugiat amet pariatur culpa tempor aliquip tempor nisi. Irure esse
      deserunt nostrud ipsum id adipisicing enim velit labore. Nulla exercitation laborum laboris
      Lorem irure sit esse nulla mollit aliquip consectetur velit
    </Panel>,
  );

  const panel = await screen.findByRole('region');

  expect(panel).toHaveStyleRule('border-radius', `${theme.borderRadius.none}`);
  expect(panel).toHaveStyleRule('border-radius', theme.borderRadius.normal, {
    media: `(min-width:${theme.breakpointValues.tablet})`,
  });
});
