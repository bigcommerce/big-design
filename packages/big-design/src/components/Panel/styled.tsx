import { theme as defaultTheme } from '@bigcommerce/big-design-theme';
import styled from 'styled-components';

import { Box } from '../Box';
import { StyleableH2 } from '../Typography/private';

export const StyledPanel = styled(Box)`
  border-radius: ${({ theme }) => theme.borderRadius.none};

  ${({ theme }) => theme.breakpoints.tablet} {
    border-radius: ${({ theme }) => theme.borderRadius.normal};
  }
`;

export const StyledH2 = styled(StyleableH2)`
  flex-grow: 1;
  flex-shrink: 1;

  & ~ .bd-button {
    width: auto;
    margin-top: 0;
  }
`;

StyledPanel.defaultProps = { theme: defaultTheme };
StyledH2.defaultProps = { theme: defaultTheme };
