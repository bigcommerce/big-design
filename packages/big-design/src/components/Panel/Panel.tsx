import React, { ComponentPropsWithoutRef, forwardRef, isValidElement, memo, Ref } from 'react';

import { excludeMarginProps, MarginProps } from '../../helpers';
import { excludePaddingProps } from '../../helpers/paddings/paddings';
import { warning } from '../../utils';
import { Badge, BadgeProps } from '../Badge/Badge';
import { Box } from '../Box';
import { Button, ButtonProps } from '../Button';
import { Dropdown, DropdownProps } from '../Dropdown';
import { Flex } from '../Flex';
import { Lozenge, LozengeProps } from '../Lozenge';
import { Text } from '../Typography';

import { StyledH2, StyledPanel } from './styled';

interface PrivateProps {
  forwardedRef: Ref<HTMLDivElement>;
}

export interface PanelActionProps extends Omit<ButtonProps, 'children' | 'mobileWidth'> {
  text?: string;
}

export interface PanelDropdownProps extends Omit<DropdownProps, 'toggle'> {
  toggle: PanelActionProps;
}

export type PanelAction = PanelActionProps | PanelDropdownProps;

export interface PanelProps extends ComponentPropsWithoutRef<'div'>, MarginProps {
  children?: React.ReactNode;
  description?: React.ReactNode;
  header?: string;
  headerId?: string;
  action?: PanelAction;
  badge?: BadgeProps;
  lozenge?: LozengeProps;
}

const Action = (action: PanelAction) => {
  if ('toggle' in action) {
    const { toggle, ...dropdownProps } = action;
    const { text, ...buttonProps } = toggle;

    return (
      <Dropdown
        {...dropdownProps}
        toggle={
          <Button {...excludeMarginProps(buttonProps)} mobileWidth="auto">
            {text}
          </Button>
        }
      />
    );
  }

  const { text, ...buttonProps } = action;

  return (
    <Button {...excludeMarginProps(buttonProps)} mobileWidth="auto">
      {text}
    </Button>
  );
};

export const RawPanel: React.FC<PanelProps & PrivateProps> = memo(({ forwardedRef, ...props }) => {
  const filteredProps = excludePaddingProps(props);
  const { action, children, description, header, headerId, badge, lozenge, ...rest } =
    filteredProps;

  const renderHeader = () => {
    if (!header && !action) {
      return null;
    }

    if (typeof header !== 'string') {
      return null;
    }

    return (
      <Flex alignItems="center" flexDirection="row" flexGap="0.5rem" justifyContent="space-between">
        {Boolean(header) && (
          <Flex alignItems="center" flexDirection="row" flexGap="0.5rem">
            <StyledH2 id={headerId} marginBottom="none">
              {header}
            </StyledH2>
            {lozenge && <Lozenge {...lozenge} />}
            {badge && <Badge {...badge} />}
          </Flex>
        )}
        {action && <Action {...action} />}
      </Flex>
    );
  };

  const renderDescription = () => {
    if (!description) {
      return null;
    }

    if (typeof description === 'string') {
      return (
        <Text color="secondary60" marginBottom="none" marginTop={header ? 'xxSmall' : 'none'}>
          {description}
        </Text>
      );
    }

    if (isValidElement(description)) {
      return description;
    }

    warning('description must be either a string or a ReactNode.');
  };

  return (
    <StyledPanel
      marginBottom="medium"
      {...rest}
      backgroundColor="white"
      padding={{ mobile: 'medium', tablet: 'xLarge' }}
      ref={forwardedRef}
      shadow="raised"
    >
      {renderHeader()}
      {renderDescription()}
      {children !== undefined && <Box marginTop="medium">{children}</Box>}
    </StyledPanel>
  );
});

export const Panel = forwardRef<HTMLDivElement, PanelProps>(
  ({ className, style, ...props }, ref) => <RawPanel {...props} forwardedRef={ref} />,
);

Panel.displayName = 'Panel';
