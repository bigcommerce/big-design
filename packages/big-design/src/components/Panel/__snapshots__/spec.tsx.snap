// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render panel 1`] = `
.c0 {
  margin-bottom: 1rem;
  box-sizing: border-box;
  background-color: #FFFFFF;
  border-radius: 0.25rem;
  box-shadow: 0px 1px 6px rgba(49,52,64,0.2);
}

.c2 {
  margin-top: 1rem;
  box-sizing: border-box;
}

.c1 {
  border-radius: 0;
}

@media (min-width:0px) {
  .c0 {
    padding: 1rem;
  }
}

@media (min-width:720px) {
  .c0 {
    padding: 1.5rem;
  }
}

@media (min-width:720px) {
  .c1 {
    border-radius: 0.25rem;
  }
}

<div
  class="c0 c1"
>
  <div
    class="c2"
  >
    Dolore proident eiusmod sint est enim laboris anim minim quis ut adipisicing consectetur officia ex. Ipsum eiusmod fugiat amet pariatur culpa tempor aliquip tempor nisi. Irure esse deserunt nostrud ipsum id adipisicing enim velit labore. Nulla exercitation laborum laboris Lorem irure sit esse nulla mollit aliquip consectetur velit
  </div>
</div>
`;

exports[`render panel with only a heading and no content 1`] = `
.c5 {
  color: #313440;
  margin: 0 0 1rem;
  font-size: 1.5rem;
  font-weight: 400;
  line-height: 2rem;
  margin-bottom: 0;
}

.c0 {
  margin-bottom: 1rem;
  box-sizing: border-box;
  background-color: #FFFFFF;
  border-radius: 0.25rem;
  box-shadow: 0px 1px 6px rgba(49,52,64,0.2);
}

.c2 {
  box-sizing: border-box;
}

.c3 {
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  gap: 0.5rem;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c4 {
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  gap: 0.5rem;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c1 {
  border-radius: 0;
}

.c6 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
}

.c6 ~ .bd-button {
  width: auto;
  margin-top: 0;
}

@media (min-width:0px) {
  .c0 {
    padding: 1rem;
  }
}

@media (min-width:720px) {
  .c0 {
    padding: 1.5rem;
  }
}

@media (min-width:720px) {
  .c1 {
    border-radius: 0.25rem;
  }
}

<div
  class="c0 c1"
>
  <div
    class="c2 c3"
  >
    <div
      class="c2 c4"
    >
      <h2
        class="c5 c6"
      >
        Test Header
      </h2>
    </div>
  </div>
</div>
`;
